File: testerum-backend/testerum-runner/testerum-runner-cmdline/src/main/assembly/report_templates/common/testerum-model/report-model/model/step/call/report-step-call-arg.ts
export class ReportStepCallArg {

    constructor(public readonly name: string|null,
                public readonly content: string|null,
                public readonly type: string,
                public readonly path: string|null) {}

    static parse(input: Object): ReportStepCallArg {
        const name = input["name"];
        const content = input["content"];
        const type = input["type"];
        const path = input["path"];

        return new ReportStepCallArg(name, content, type, path);
    }
}

File: testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/report_model/ReportStepCallArg.kt
package com.testerum.runner.report_model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

class ReportStepCallArg @JsonCreator constructor(@JsonProperty("name") val name: String?,
                                                 @JsonProperty("content") val content: String?,
                                                 @JsonProperty("type") val type: String,
                                                 @JsonProperty("path") val path: String?)
